Contribute.md template

gilmrjc opened the pull request:
Added the most basic structure for a template. I left out the review process, maybe in the next pull request I can introduce one.

Close: #2

jokeyrhyme commented on CONTRIBUTING.md:
I feel like it ought to be sufficient to link to GitHub Flow: https://help.github.com/articles/github-flow/ rather than spell out how to use `git` here.

gilmrjc replied:
This section defines the workflow. It contains information like style conventions, commit message conventions, branch naming conventions. It ask you to test your code before submitting the pull request. It's better to have all this information together than just linking to the GitHub Flow documentation.

jokeyrhyme commented on CONTRIBUTING.md:
```diff
@@ -1,140 +1,296 @@
-# Contributing
+# Contributing to *PROJECT NAME*
```
I love what you've done with this file, much nicer than before!

jokeyrhyme commented on CONTRIBUTING.md:
```diff
@@ -1,140 +1,296 @@
 
-By contributing you agree to the [LICENSE](LICENSE) of this repository.
+*Note*: Be nice to your users and let them know you care about their
+contributions. Feel free to edit the text below.
```
One the ideas we discussed was making this file quicker to use for most cases. I'd be interested to see what you think about dropping such instructions for project maintainers?

jokeyrhyme replied:
Or perhaps having all instructions up the top so they can be deleted more easily?

gilmrjc replied:
You're right. I moved all the instructions to the top to remove them easier. I left a note on the *Review and approval* step because it's empty and for me it's best to keep a note there for now.

jokeyrhyme commented on CONTRIBUTING.md:
Definitely a great idea to have something like this in here

jokeyrhyme commented:
I haven't been super disciplined about this, but what do you make of this? https://scott.mn/2014/02/21/semantic_linewrapping/

gilmrjc commented:
I feel you have to be disciplined about semantic linewrapping but I like the idea. I changed the documents to follow it. The rule is now inside the contributing file as an example of coding conventions.

I moved all the instructions to the top and made a minor changes to the text.

jokeyrhyme commented:
Yay, great work! Thanks!
